python/tk_multi_publish2/api/item.py

```diff
@@ -0,0 +1,1012 @@
+        :py:attr:`~local_properties` will only be applied during the execution
+        of the current plugin (similar to python's ``threading.local`` storage).
+
+        A common scenario to consider is when you have multiple publish plugins
+        acting on the same item. You may, for example, want the ``publish_name``
+        and ``publish_version`` properties to be shared by each plugin, while
+        setting the remaining properties on each plugin instance since they will
+        be specific to that plugin's output. Example:
+
+        .. code-block:: python
+
+            # set shared properties on the item (potentially in the collector or
+            # the first publish plugin). these values will be available to all
+            # publish plugins attached to the item.
+            item.properties.publish_name = "Gorilla"
+            item.properties.publish_version = "0003"
+
+            # set specific properties in subclasses of the base file publish
+            # (this class). first publish plugin...
+            item.local_properties.publish_template = "asset_fbx_template"
+            item.local_properties.publish_type = "FBX File"
+
+            # in another publish plugin...
+            item.local_properties.publish_template = "asset_abc_template"
+            item.local_properties.publish_type = "Alembic Cache"
+
+        .. note:: If you plan to serialize your publish tree, you may run into
+            issues if you add complex or non-serializable objects to the
+            properties dictionary.
+        """
+        return self._get_local_properties()
+
+    @property
+    def name(self):
+        """The display name of the item."""
+        return self._name
+
+    @name.setter
+    def name(self, new_name):
+        """Sets a new display name for the item with the given string."""
+        self._name = new_name
+
+    @property
+    def parent(self):
+        """The item's parent :ref:`publish-api-item`."""
+        return self._parent
+
+    @property
+    def persistent(self):
+        """
+        Boolean indicator that the item should not be removed when the tree is
+        cleared.
+        """
+        return self._persistent
+
+    @persistent.setter
+    def persistent(self, is_persistent):
+        """Set the item to persistent or not.
+
+        Only top-level items can be set to persistent.
+        """
+        # It's not a crime to turn persistence off, so raise an error when
+        # actually trying it on an invalid item.
+        if is_persistent and (not self.parent or not self.parent.is_root):
+            raise sgtk.TankError(
+                "Only top-level tree items can be made persistent.")
+
+        self._persistent = is_persistent
+
+    @property
+    def properties(self):
+        """
+        A :class:`PublishData` instance where arbitrary data can be stored on
+        the item. The property itself is read-only (you can't assign a different
+        :class:`PublishData` instance.
```

Comment:
Ditch the parenthesis that opened in the previous line and never terminates. No need for this to read as an aside.